### ISSUE TITLE
TweakScalerFSbuoyancy throws System.NullReferenceException

### ISSUE THREAD

**Issue by 7ranceaddic7** [net-lisias-ksp/TweakScaleCompanion_FS#6]
The below NRE is thrown for these six (6) parts:

AirplanePlus/Parts/Utility/Skid/part/landingskid  **(!?!?!?!)**
Firespitter/Parts/Structural/FS_floatEndNose/part/FSfloatEnd
Firespitter/Parts/Structural/FS_floatEndTail/part/FSfloatEndTail
Firespitter/Parts/Structural/FS_floatGearbay/part/FSfloatGearbay
Firespitter/Parts/Structural/FS_floatStraight/part/FSfloatStraight
Firespitter/Parts/Structural/FS_floatStrut/part/FSfloatStrut

[Full Log.](https://drive.google.com/file/d/1TwTVim7rcSPgdv5BhL19yVVmUIaibs9u/view?usp=drivesdk)

======== Begin ========
[LOG 23:27:47.643] PartLoader: Compiling Part 'Firespitter/Parts/Structural/FS_floatEndNose/part/FSfloatEnd'
[ERR 23:27:47.662] Module TweakScalerFSbuoyancy threw during OnLoad: System.NullReferenceException: Object reference not set to an instance of an object
  at TweakScaleCompanion.FS.Buoyancy.TweakScalerFSbuoyancy.InitInternalData () [0x00042] in <9042ad2d21684961ba52e9c0c294474c>:0 
  at TweakScaleCompanion.FS.Buoyancy.TweakScalerFSbuoyancy.OnLoad (ConfigNode node) [0x00056] in <9042ad2d21684961ba52e9c0c294474c>:0 
  at PartModule.Load (ConfigNode node) [0x001ab] in <55ba45dc3a43403382024deac8dcd0be>:0 

[LOG 23:27:47.693] PartLoader: Part 'Firespitter/Parts/Structural/FS_floatEndNose/part/FSfloatEnd' has no database record. Creating.
[LOG 23:27:47.707] DragCubeSystem: Creating drag cubes for part 'FSfloatEnd'
======== End ========

**Comment by Lisias** [comment]
Hi. Thanks for the report. I will copy&paste what I said on another issue for the sake of completude:

> What happens is that FSBuoyancy is somewhat uncollaborative and I need to override its UI in order to made it to behave.
>
> However, the Part Module's life cycle on KSP was not really intended to have dependencies from other modules - it's assumed that everything needed by the PartModule is already loaded and ready to be used, what's not the case when you want to access some other Part Module - they are initialized and started on a random order (not exactly, but we need to assume it's random). So I coded some stunts to cope with that, delaying the Helper initialisation until the FSBuyoancy is ready to be instrumented.
>
> In the rush to fix the TweakScale Sanity Checks, I'm pretty sure I broke something needed by the Helper, as this poor bastard depends of having TweakScale and FSBuyoancy initialized and started to carry on.

I think it's something silly, I should had changed something needed by the Helper without being aware of it. I'll try to have it fixed by the night.

**Comment by 7ranceaddic7** [comment]
Sorry, I didn't find that explanation or I just would have bumped that issue.

Don't sweat it too much.  It's only parts that are used on rare occasions.

**Comment by 7ranceaddic7** [comment]
I may have found the thing you broke.

**Comment by 7ranceaddic7** [comment]
Forgive me if I'm being dumb (probably), but do you know about this? Could there be unintended scaling going on in the bouyancyRange or VerticalOffset??

FSbuoyancy
This module kills the parts stock buoyancy and replaces it with a custom float code.

MODULE
{
name = FSbuoyancy
buoyancyForce = 12.0 // the force applied to lift the part, scaled by depth according to buoyancyRange
buoyancyRange = 1.0 // the max depth at which the buoyancy will be scaled up. at this depth, the force applied is equal to 
                                        buyoancyForce. At 0 depth, the force is 0
buoyancyVerticalOffset = 0.1 // how high the part rides on the water in meters. Not a position offset inside the part. This will 
                                                   be applied in the global axis regardless of part rotation. Think iceberg/styrofoam.
// for subs you could play with that value to make it think you are at sea level buoyancywise at an entirely different depth. You could even change this value at run time to create the whole up/down code (Using your own code I mean)
maxVerticalSpeed = 0.2 // the max speed vertical speed at which there will be a lifting force applied. Reduces bobbing.
dragInWater = 1.5 // when in water, apply drag to slow the craft down. Stock drag is 3.
debugMode = True // enables buoyancy control context menus
waterImpactTolerance = 125 // For seaplanes, any part in the water during take-off should have a resistance high enough to 
                                                  withstand take off speeds.
//forcePointName = // you can set an alternate transform name to set up multi-point float. Optional
splashFXEnabled = True // on by default, creates splashes even deep underwater currently...

}

**Comment by Lisias** [comment]
> Forgive me if I'm being dumb (probably), but do you know about this? Could there be unintended scaling going on in the bouyancyRange or VerticalOffset??

Not exactly. It doesn't kills the original module, it "hides" it and play proxy for it, so when user changes something, the proxy takes the change, update the module as it was the user that was doing it, and then calls TweakScale to do it's magic.

What I borked on this "proxy" (technically, a Decorator as described on the Design Patterns Bibble) was that I made some weak assumptions to burn some time into the Alpha stage, and these assumptions blow up on my face when I updated TweakScale due a change on the Module's Initialisation criteria that happened on KSP >= 1.8.

It's fixed by now, but I didn't had the time to publish the new release on the Working Days.
